packages/apps/plugins/plugin-map/src/components/MapDetectLocations.ts

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { CollectionType, TableType, type MapType } from '@braneframe/types';
+import { getTypename, type DynamicSchema } from '@dxos/echo-schema';
+import { Filter, getSpace, useQuery } from '@dxos/react-client/echo';
+
+import { type Marker } from './MapControl';
+
+/**
+ * Find table rows that are in the same collection as the map.
+ * Return the ones with `latitude` and `longitude` fields,
+ * converted to `Marker`s.
+ *
+ * NOTE: This collects rows from all the collections the map is in,
+ * even if there's more than one. At the time of writing this behavior
+ * won't be exercised since being included in multiple collections
+ * is disabled in the UI.
+ *
+ * If we enable being in more than one collection again we should
+ * reconsider what behavior we actually want in that case.
+ */
+export const useMapDetectLocations = (map: MapType): Marker[] => {
+  const space = getSpace(map);
+
+  const parentCollections = useQuery<CollectionType>(
+    space,
+    [
+      Filter.schema(CollectionType),
+      (collection: any) => {
+        if (!(collection instanceof CollectionType)) {
+          return false;
+        }
+        return collection.objects.some((obj) => obj?.id === map.id);
+      },
+    ],
+    undefined,
+    [space],
+  );
+
+  const siblingIds = new Set(
+    parentCollections.flatMap((collection) => collection.objects.filter(hasId).map((obj) => obj.id)),
+  );
+
+  const siblingTables = useQuery<TableType>(
+    space,
+    [
+      Filter.schema(TableType),
+      (table: any) => siblingIds.has(table.id),
+      (table: any) => {
+        if (!(table instanceof TableType)) {
+          return false;
+        }
+        return table.schema ? schemaHasLatitudeAndLongitude(table.schema) : false;
+      },
+    ],
+    undefined,
+    [JSON.stringify(Array.from(siblingIds))],
+  );
+
+  const schemaIds: string[] = siblingTables.flatMap((table) => (table.schema ? [table.schema.id] : []));
+
+  const rows = useQuery(
+    // Short circuit if there are no `schemaIds` to find rows for:
+    schemaIds.length === 0 ? undefined : space,
+    // PERFORMANCE: checks every object.
+    // Can't pass `Filter.or`s to `useQuery` because we need to a dynamic number of them.
```

Comment:
```suggestion
    // TODO(seagreen): This checks every object.
    //   Can't pass list of `Filter.or`s to `useQuery` because we need to a dynamic number of them.
```

Reply:
@dmaretskyi we need to consider how to combine these filters in a way where there's just one root so that the react deps array size doesn't change